homeassistant/components/camera/xeoma.py

```diff
@@ -0,0 +1,112 @@
+"""
+Support for Xeoma Cameras.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/camera.xeoma/
+"""
+import asyncio
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.camera import (
+    PLATFORM_SCHEMA, Camera)
+from homeassistant.const import (
+    CONF_NAME, CONF_USERNAME, CONF_PASSWORD, CONF_HOST)
+from homeassistant.helpers import config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+REQUIREMENTS = ['pyxeoma==1.2']
+
+CONF_CAMERAS = 'cameras'
+CONF_HIDE = 'hide'
+CONF_IMAGE_NAME = 'image_name'
+CONF_NEW_VERSION = 'new_version'
+
+CAMERAS_SCHEMA = vol.Schema({
+    vol.Required(CONF_IMAGE_NAME): cv.string,
+    vol.Optional(CONF_HIDE, default=False): cv.boolean,
+    vol.Optional(CONF_NAME): cv.string
+}, required=False)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_CAMERAS, default={}):
+        vol.Schema(vol.All(cv.ensure_list, [CAMERAS_SCHEMA])),
+    vol.Optional(CONF_PASSWORD): cv.string,
+    vol.Optional(CONF_USERNAME): cv.string,
+    vol.Optional(CONF_NEW_VERSION, default=True): cv.boolean
+})
+
+
+@asyncio.coroutine
+# pylint: disable=unused-argument
+def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
+    """Discover and setup Xeoma Cameras."""
+    host = config[CONF_HOST]
+    login = config[CONF_USERNAME] if CONF_USERNAME in config else None
```

Comment:
Use `login = config.get(CONF_USERNAME)`. This way it will be `None` if no username was set in the configuration.

Reply:
Thanks. Done